data/templates/ipsec/swanctl.conf.j2

```diff
@@ -87,7 +87,13 @@ secrets {
         id-{{ gen_uuid }} = "{{ id }}"
 {%             endfor %}
 {%         endif %}
+{%         if psk_config.secret_type is vyos_defined('base64') %}
+        secret = 0s{{ psk_config.secret }}
+{%         elif psk_config.secret_type is vyos_defined('plaintext') %}
         secret = "{{ psk_config.secret }}"
+{%         else %}
```

Comment:
If `secret_type` has a `defaultValue` in XML there is no need for a `else` code path.

Reply:
I wanted to play it safe and make a fallback.